torch/distributed/_functional_collectives_impl.py

```diff
@@ -117,11 +117,6 @@ def _wait_reg_dec(ptr, wait_reg):
 def _register_tensor_wrapper(tensor) -> None:
     if native_funcol_enabled():
         # Tensor storage -> work mapping is maintained in C++
-        weakref.finalize(
```

Comment:
hmm, what would have been the design intent behind automating a wait() call on work destruction?

I tend to think this change is ok, because I assume users would typically keep their ACT object alive long enough to later use it to wait.  As long as it doesn't break program exit somehow, sgtm.